upup/pkg/fi/secrets/clientset_secretstore.go

```diff
@@ -221,5 +236,8 @@ func (c *ClientsetSecretStore) createSecret(s *fi.Secret, name string) (*kops.Ke
 		PrivateMaterial: s.Data,
 	})
 
+	if replace {
+		return c.clientset.Keysets(c.namespace).Update(keyset)
```

Comment:
Challenge here:I think Update fails if the secret doesn't exist.  You would have to read it first.